tt_eager/tt_dnn/op_library/unpad/unpad_op.cpp

```diff
@@ -18,18 +18,41 @@ namespace tt {
 
 namespace tt_metal {
 
+uint32_t get_tiled_start_offset(const Tensor &input_tensor,
+                        const Shape &output_tensor_start
+ ) {
+    uint32_t num_input_pages = input_tensor.volume() / (TILE_HW);
+    uint32_t num_pages_last_two_dim = num_input_pages / (input_tensor.shape()[0] * input_tensor.shape()[1]);
+    uint32_t num_pages_width = num_input_pages / (input_tensor.shape()[0] * input_tensor.shape()[1] * (input_tensor.shape()[2]/TILE_HEIGHT));
+    uint32_t num_pages_height = num_input_pages / (input_tensor.shape()[0] * input_tensor.shape()[1] * (input_tensor.shape()[3]/TILE_WIDTH));
+    uint32_t start_offset = ((output_tensor_start[0] * output_tensor_start[1] ) * num_pages_last_two_dim +
+                                output_tensor_start[2]/TILE_HEIGHT*num_pages_width + output_tensor_start[3]/TILE_WIDTH);
+    return start_offset;
```

Comment:
you should remove the hardcoding to 4D here.
Use -2 and -1 for last two dims.
Multiply all elements together except last two for where you have [0] * [1]

These don't seem to correctly take into account non-last two dim start offsets. Is this expected?

Reply:
Thanks Austin, ya I changed to account for upper dims and generalized it.